src/main/java/org/jboss/arquillian/extension/jacoco/client/filter/AndFilter.java

```diff
@@ -16,13 +16,18 @@
     @Override
     public boolean include(T object)
     {
+        boolean include = false;
         for (Filter<T> f : filters)
         {
             if (!f.include(object))
             {
+                if (f.toString().contains("IncludeRegExpPaths") && include) {
```

Comment:
I don't think this is ensuring correctness of this scenario. We probably have to have following structure instead:

I - set of include patterns
E - set of exclude patterns
```
E = e1 || e2 || .... || eN
I = i1 || i2 || ... || iN
```
the asset is included if  `!E && I`

Thus I think we should not only rely on `AndFilter` but improve the logic based on what this test exposed.

In addition, the generic filter (`AndFilter` in this case) shouldn't assume anything about the underlying implementations it's delegating to.